CMakeLists.txt

```diff
@@ -179,13 +179,18 @@ function (ADD_CHECK_CXX_FLAG _CXXFLAGS _CACHE_VAR _FLAG)
 endfunction ()
 
 if (MSVC)
+    # CMAKE does not have a built-in option for setting the CRT, so override the default flags.
+    # NOTE: doing it this way avoids a linker warning about one directive overriding another
+    set(CMAKE_CXX_FLAGS_DEBUG "/MTd /Zi /Ob0 /Od /RTC1")
+    set(CMAKE_CXX_FLAGS_RELEASE "/MT /O2 /Ob2 /DNDEBUG")
+
     set(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} /utf-8 /permissive- /Zc:externConstexpr /EHsc /WX")
     set(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} /wd4244") # C4244: 'conversion_type': conversion from 'type1' to 'type2', possible loss of data
     set(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} /wd4068") # C4068: unknown pragma
 
     add_definitions(-D_CRT_SECURE_NO_WARNINGS)
     add_definitions(-D_SCL_SECURE_NO_WARNINGS)
-    if ((CMAKE_VS_PLATFORM_NAME STREQUAL "Win32") OR (CMAKE_VS_PLATFORM_NAME STREQUAL "x64"))
+    if (CMAKE_SIZEOF_VOID_P EQUAL 8)
```

Comment:
Arm64 doesn't have sse nor avx

Reply:
So what can I put here instead to fix this?

Reply:
?

Reply:
`CMAKE_SYSTEM_PROCESSOR`?

Reply:
The way cmake expects to work for Windows and MSVC is that you launch the correct VS environment for your target platform then run cmake.

`CMAKE_SYSTEM_PROCESSOR` always returns AMD64 for both an x86 and AMD64 visual studio development environment.

Do you have any other alternatives?

Reply:
If not then I will merge this as we don't yet support Windows on ARM, it can be changed when we do.

Reply:
But we do and this (current cmake) is how it is achieved.

Reply:
How about this?

`if (CMAKE_SIZEOF_VOID_P EQUAL 8 AND CMAKE_SYSTEM_PROCESSOR STREQUAL "AMD64")`

Reply:
You mentioned earlier that the variable was not set to expected value when using x86_64 and x86, does it work for amd64?

Reply:
Do you mean ARM64? It always returns AMD64 for Win32 and x64 which is why the pointer size check is also needed to differentiate between the platforms. I would expect it to return ARM64 if you were targeting ARM, but I have no way of checking this.

Reply:
Yeah, misread the value there. Checking should be trivial, create a dummy cmake project and have it print the value and pass on the relevant generator, both msbuild and ninja.

Reply:
I think I need the ARM cross compiling tools for Visual Studio for me to try this and I don't want to install that as it is quite large.